endpoint.go

```diff
@@ -122,17 +121,39 @@ type InEndpoint struct {
 	*endpoint
 }
 
-// Read reads data from an IN endpoint.
+// Read reads data from an IN endpoint. Read returns number of bytes obtained
+// from the endpoint. Read may return non-zero length even if
+// the returned error is not nil (partial read).
 func (e *InEndpoint) Read(buf []byte) (int, error) {
-	return e.transfer(buf)
+	return e.transfer(context.Background(), buf)
+}
+
+// ReadContext reads data from an IN endpoint. Read returns number of bytes
```

Comment:
nit: s/Read/ReadContext/, also in the next line.

Reply:
done